internal/nginx/config/generator.go

```diff
@@ -1,19 +1,45 @@
 package config
 
 import (
+	"path/filepath"
+
+	"github.com/nginxinc/nginx-kubernetes-gateway/internal/nginx/file"
 	"github.com/nginxinc/nginx-kubernetes-gateway/internal/state/dataplane"
 )
 
 //go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . Generator
 
-// Generator generates NGINX configuration.
+const (
+	// configFolder is the folder where NGINX configuration files are stored.
+	configFolder = "/etc/nginx"
+
+	// httpFolder is the folder where NGINX HTTP configuration files are stored.
+	httpFolder = configFolder + "/conf.d"
```

Comment:
For safety, we should probably use `filepath.Join` to create these paths, rather than just concatenation.

Reply:
using the function will make it impossible to use a constant there.

since it is a constant and we're not dealing with user input, perhaps we can trust those values?

otherwise, I will need to make them as variables - either global or part of `GeneratorImpl`.

Reply:
Understood, that's fine.